src/Quantum.jl

```diff
@@ -335,7 +335,21 @@ end
 
 Returns the adjoint of a [`Quantum`](@ref) Tensor Network; i.e. the conjugate Tensor Network with the inputs and outputs swapped.
 """
-Base.adjoint(tn::AbstractQuantum) = adjoint!(deepcopy(tn))
+function Base.adjoint(tn::AbstractQuantum)
+    tn = conj(tn)
+
+    # update site information
+    oldsites = copy(Quantum(tn).sites)
+    empty!(Quantum(tn).sites)
+    for (site, index) in oldsites
+        addsite!(tn, site', index)
+    end
+
+    # rename inner indices
+    replace!(tn, map(i -> i => Symbol(i, "'"), inds(tn; set=:virtual)))
+
+    return tn
+end
```

Comment:
Can you move this code to a new `adjoint_sites` method and deduplicate the code?